cromwellApiClient/src/main/scala/cromwell/api/CromwellClient.scala

```diff
@@ -0,0 +1,127 @@
+package cromwell.api
+
+import java.net.URL
+
+import akka.http.scaladsl.Http
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.{HttpEntity, _}
+import akka.http.scaladsl.unmarshalling.{Unmarshal, Unmarshaller}
+import akka.stream.ActorMaterializer
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+import akka.util.ByteString
+import cromwell.api.model._
+import spray.json._
+
+import scala.collection.immutable
+import scala.concurrent.{ExecutionContext, Future}
+import cromwell.api.CromwellClient._
+
+import scala.util.{Failure, Success, Try}
+
+class CromwellClient(val cromwellUrl: URL, val apiVersion: String)(implicit actorSystem: ActorSystem, materializer: ActorMaterializer) {
+
+  lazy val submitEndpoint = s"$cromwellUrl/api/workflows/$apiVersion"
+  // Everything else is a suffix off the submit endpoint:
+  lazy val batchSubmitEndpoint = s"$submitEndpoint/batch"
+  private def workflowSpecificEndpoint(workflowId: WorkflowId, endpoint: String) = s"$submitEndpoint/$workflowId/$endpoint"
+  def abortEndpoint(workflowId: WorkflowId) = workflowSpecificEndpoint(workflowId, "abort")
+  def statusEndpoint(workflowId: WorkflowId) = workflowSpecificEndpoint(workflowId, "status")
+  def metadataEndpoint(workflowId: WorkflowId) = workflowSpecificEndpoint(workflowId, "metadata")
+  lazy val backendsEndpoint = s"$submitEndpoint/backends"
+
+  import model.CromwellStatusJsonSupport._
+  import model.CromwellBackendsJsonSupport._
+
+  def submit(workflow: WorkflowSubmission)(implicit ec: ExecutionContext): Future[SubmittedWorkflow] = {
+    val sourceBodyParts = Map(
+      "wdlSource" -> Option(workflow.wdl),
+      "workflowInputs" -> workflow.inputs,
+      "workflowOptions" -> insertSecrets(workflow.options, workflow.refreshToken)
+    ) collect { case (name, Some(source)) => Multipart.FormData.BodyPart(name, HttpEntity(MediaTypes.`application/json`, ByteString(source))) }
+
+    val zipBodyParts = Map(
+      "wdlDependencies" -> workflow.zippedImports
+    ) collect { case (name, Some(file)) => Multipart.FormData.BodyPart.fromPath(name, MediaTypes.`application/zip`, file.path) }
+
+    val multipartFormData = Multipart.FormData((sourceBodyParts ++ zipBodyParts).toSeq : _*)
+    val requestEntity = multipartFormData.toEntity()
+
+    makeRequest[CromwellStatus](HttpRequest(HttpMethods.POST, submitEndpoint, immutable.Seq.empty[HttpHeader], requestEntity)) map { status =>
+      SubmittedWorkflow(WorkflowId.fromString(status.id), cromwellUrl, workflow)
+    }
+  }
+
+  def submitBatch(workflow: WorkflowBatchSubmission)(implicit ec: ExecutionContext): Future[Seq[SubmittedWorkflow]] = {
```

Comment:
I'd prefer concrete return types instead of `Seq` (However I've given up complaining on incoming parameters)